nifi-extension-bundles/nifi-slack-bundle/nifi-slack-processors/src/main/java/org/apache/nifi/processors/slack/ConsumeSlack.java

```diff
@@ -255,38 +259,54 @@ private List<ConsumeChannel> createChannels(final ProcessContext context, final
             .filter(s -> !s.isEmpty())
             .forEach(channels::add);
 
-        // Fetch all channel ID's to have a name/ID channel mapping
-        Map<String, String> channelMapping = client.fetchChannelIds();
+        Map<String, String> channelMapping = new HashMap<>();
+
+        if (channelIdsProvidedOnly(channels)) {
+            //resolve the channel names by the specified channel IDs
+            for (String channelId : channels) {
+                String channelName = client.fetchChannelName(channelId);
+                getLogger().info("Resolved Channel ID {} to name {}", channelId, channelName);
```

Comment:
I am not sure the possible number of channels but it does not seem right to have so many log statement on `info` level here, line  285 and line 294.